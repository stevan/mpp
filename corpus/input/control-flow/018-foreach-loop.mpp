for my $i (1..10) {
    print($i);
}