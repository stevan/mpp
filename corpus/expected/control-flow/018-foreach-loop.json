[
  {
    "type": "Foreach",
    "variable": {
      "type": "Variable",
      "name": "$i"
    },
    "declarator": "my",
    "listExpr": {
      "type": "BinaryOp",
      "operator": "..",
      "left": {
        "type": "Number",
        "value": "1"
      },
      "right": {
        "type": "Number",
        "value": "10"
      }
    },
    "block": [
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "Variable",
            "name": "$i"
          }
        ]
      }
    ]
  }
]